client/catapult/CMakeGlobalSettings.cmake

```diff
@@ -160,6 +160,8 @@ elseif("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
 	# - Wno-padded: allow compiler to automatically pad data types for alignment
 	# - Wno-switch-enum: do not require enum switch statements to list every value (this setting is also incompatible with GCC warnings)
 	# - Wno-weak-vtables: vtables are emitted in all translation units for virtual classes with no out-of-line virtual method definitions
+	# - Wno-unsafe-buffer-usage: allow unsafe buffer usage https://reviews.llvm.org/D137379
```

Comment:
what code is triggering this?

Reply:
There are multiple places but here is a few.  Iterator doesn't make sense to me.

```
/catapult-src/client/catapult/src/catapult/utils/HexFormatter.h:103:67: error: unsafe pointer arithmetic [-Werror,-Wunsafe-buffer-usage]
  103 |                         for (auto iter = formatter.m_begin; formatter.m_end != iter; ++iter) {
      |                                                                                        ^~~~
/catapult-src/client/catapult/src/catapult/utils/HexFormatter.h:106:39: error: unsafe pointer arithmetic [-Werror,-Wunsafe-buffer-usage]
  106 |                                 if (0 != formatter.m_separator && iter + 1 != formatter.m_end)
      |                                                                   ^~~~
  ```
  
  I understand why these were triggered
  ```
  /catapult-src/client/catapult/src/catapult/utils/PathUtils.h:39:27: error: unsafe pointer arithmetic [-Werror,-Wunsafe-buffer-usage]
   39 |                 while (*pEnd != '\0') ++pEnd;
      |                                         ^~~~
/catapult-src/client/catapult/src/catapult/utils/PathUtils.h:47:26: error: unsafe pointer arithmetic [-Werror,-Wunsafe-buffer-usage]
   47 |                 for (const auto* pCh = pEnd - 1; pCh >= fullPath; --pCh) {
      |                                        ^~~~
/catapult-src/client/catapult/src/catapult/utils/PathUtils.h:47:55: error: unsafe pointer arithmetic [-Werror,-Wunsafe-buffer-usage]
   47 |                 for (const auto* pCh = pEnd - 1; pCh >= fullPath; --pCh) {
      |    
  ```